cmd/attest_test.go

```diff
@@ -0,0 +1,374 @@
+			RootCmd.SetArgs(debugArgs)
+			if err := RootCmd.Execute(); err == nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestMetadataPass(t *testing.T) {
+	var dummyInstance = Instance{ProjectID: "test-project", ProjectNumber: "1922337278274", Zone: "us-central-1a", InstanceID: "12345678", InstanceName: "default"}
+	mock, err := NewMetadataServer(dummyInstance)
+	if err != nil {
+		t.Error(err)
+	}
+	defer mock.Stop()
+	instanceInfo, err := getInstanceInfoFromMetadata()
+	if err != nil {
+		t.Error(err)
+	}
+	if instanceInfo.ProjectId != dummyInstance.ProjectID {
+		t.Errorf("metadata.ProjectID() = %v, want %v", instanceInfo.ProjectId, dummyInstance.ProjectID)
+	}
+	projectNumber, err := strconv.ParseUint(dummyInstance.ProjectNumber, 10, 64)
+	if err != nil {
+		t.Error(err)
+	}
+	if instanceInfo.ProjectNumber != projectNumber {
+		t.Errorf("metadata.NumericProjectID() = %v, want %v", instanceInfo.ProjectNumber, projectNumber)
+	}
+	if instanceInfo.InstanceName != dummyInstance.InstanceName {
+		t.Errorf("metadata.InstanceName() = %v, want %v", instanceInfo.InstanceName, dummyInstance.InstanceName)
+	}
+	instanceID, err := strconv.ParseUint(dummyInstance.InstanceID, 10, 64)
+	if err != nil {
+		t.Error(err)
+	}
+	if instanceInfo.InstanceId != instanceID {
+		t.Errorf("metadata.InstanceID() = %v, want %v", instanceInfo.InstanceId, instanceID)
+	}
+	if instanceInfo.Zone != dummyInstance.Zone {
+		t.Errorf("metadata.Zone() = %v, want %v", instanceInfo.Zone, dummyInstance.Zone)
+	}
+}
+
+func TestAttestWithGCEAK(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+	secretFile1 := makeOutputFile(t, "attest")
+	defer os.RemoveAll(secretFile1)
+	var template = map[string]tpm2.Public{
+		"rsa": GCEAKTemplateRSA(),
+		"ecc": GCEAKTemplateECC(),
+	}
+	tests := []struct {
+		name    string
+		nonce   string
+		keyAlgo string
+	}{
+		{"gceAK:RSA", "1234", "rsa"},
+		{"gceAK:ECC", "1234", "ecc"},
+	}
+	for _, op := range tests {
+		t.Run(op.name, func(t *testing.T) {
+			data, err := template[op.keyAlgo].Encode()
+			if err != nil {
+				t.Fatalf("failed to encode GCEAKTemplateRSA: %v", err)
+			}
+			err = setGCEAKTemplate(t, rwc, op.keyAlgo, data)
+			if err != nil {
+				t.Error(err)
+			}
+			defer tpm2.NVUndefineSpace(rwc, "", tpm2.HandlePlatform, tpmutil.Handle(getIndex[op.keyAlgo]))
+
+			var dummyInstance = Instance{ProjectID: "test-project", ProjectNumber: "1922337278274", Zone: "us-central-1a", InstanceID: "12345678", InstanceName: "default"}
+			mock, err := NewMetadataServer(dummyInstance)
+			if err != nil {
+				t.Error(err)
+			}
+			defer mock.Stop()
+
+			RootCmd.SetArgs([]string{"attest", "--nonce", op.nonce, "--key", "gceAK", "--algo", op.keyAlgo, "--output", secretFile1, "--format", "binarypb"})
+			if err := RootCmd.Execute(); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestAttestSevSnpPass(t *testing.T) {
```

Comment:
I'd like to see a test that the verify command works as well when the attestation includes a SevSnp report.